internal/rewrite_action.go

```diff
@@ -156,22 +156,8 @@ func (t *ImageTemplate) Apply(originalImage name.Reference, rules *rewrite.Rules
 	var rewrites []*RewriteAction
 
 	// Tag or Digest
-	if t.TagTemplate != "" {
+	if t.TagTemplate != "" || t.DigestTemplate != "" {
 		tagged = true
-		if rules.Tag != "" && rules.Tag != originalImage.Identifier() {
-			rewrites = append(rewrites, &RewriteAction{
-				Path:  t.TagTemplate,
-				Value: rules.Tag,
-			})
-		}
-	} else if t.DigestTemplate != "" {
-		tagged = true
-		if rules.Digest != "" && rules.Digest != originalImage.Identifier() {
-			rewrites = append(rewrites, &RewriteAction{
-				Path:  t.DigestTemplate,
```

Comment:
If the digest is different we should not push it.

Reply:
You are right, that's what I meant actually, that we do not push an image with a different digest so this function had functionality that now is in practice unreachable. Note that I have not added the protection, just removed the code that performed that action.

Are you ok with the removal or would you prefer me to put it back?

Reply:
I think it's good.